extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/resources/SqlStatementResource.java

```diff
@@ -620,15 +653,30 @@ private MSQControllerTask getMSQControllerTaskOrThrow(String queryId, String cur
                                                        .getQuery()
                                                        .getContext()
                                                        .get(MSQTaskQueryMaker.USER_KEY));
-    if (currentUser == null || !currentUser.equals(queryUser)) {
-      throw new ForbiddenException(StringUtils.format(
-          "The current user[%s] cannot view query id[%s] since the query is owned by user[%s]",
-          currentUser,
-          queryId,
-          queryUser
-      ));
+
+    String currentUser = authenticationResult.getIdentity();
+
+
+    if (currentUser != null && currentUser.equals(queryUser)) {
+      return msqControllerTask;
+    }
+
+    Access access = AuthorizationUtils.authorizeAllResourceActions(
+        authenticationResult,
+        Collections.singletonList(new ResourceAction(Resource.STATE_RESOURCE, forAction)),
+        authorizerMapper
+    );
+
+    if (access.isAllowed()) {
+      return msqControllerTask;
     }
-    return msqControllerTask;
+
+    throw new ForbiddenException(StringUtils.format(
+        "The current user[%s] cannot view query id[%s] since the query is owned by user[%s]",
+        currentUser,
+        queryId,
+        queryUser
+    ));
```

Comment:
```suggestion
    throw new ForbiddenException(StringUtils.format(
        "The current user[%s] cannot view query id[%s] since the query is owned by another user",
        currentUser,
        queryId
    ));
```
We should probably not expose the query user to the current user

Reply:
Good catch